index.ts

```diff
@@ -31,27 +31,23 @@ export default function diff(
 			continue;
 		}
 		const newObjKey = newObj[key];
-		const areObjects =
-			typeof objKey === "object" && typeof newObjKey === "object";
+		const areObjects = typeof objKey === o && typeof newObjKey === o;
 		if (
 			objKey &&
 			newObjKey &&
 			areObjects &&
 			!richTypes[Object.getPrototypeOf(objKey).constructor.name] &&
-			(options.cyclesFix ? !_stack.includes(objKey) : true)
+			(!opts.cyclesFix || !_stack.includes(objKey))
 		) {
 			const nestedDiffs = diff(
 				objKey,
 				newObjKey,
-				options,
-				options.cyclesFix ? _stack.concat([objKey]) : []
+				opts,
+				opts.cyclesFix ? _stack.concat([objKey]) : []
 			);
 			diffs.push.apply(
 				diffs,
-				nestedDiffs.map((difference) => {
-					difference.path.unshift(path);
-					return difference;
-				})
+				nestedDiffs.map((difference) => (difference.path.unshift(path), difference))
```

Comment:
> I do not think the unshift change will work appropriately due to unshifts return value.

actually, this change has the same behavior as the other one but with less bytes. It uses the [comma operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comma_Operator). The return of `unshift` doesn't matter as the return of `map`'s callback will always be `difference` (just like the current version)

Reply:
but I was not taking minifiers in mind. This change would have not difference in the end if you use such tool :smile:

Reply:
I did not even notice that 🙃.
I am focusing on the minified output because on the web, where size is most important,  most people use minifiers. I do care about size for Node. However, I am not trying to micro-optimize it as much.

Reply:
makes sense.